examples_gbs/run_tutorial_max_clique.py

```diff
@@ -0,0 +1,158 @@
+is the *binding interaction graph* representing the spatial compatibility of atom pairs in a
+protein-molecule complex. Cliques in this graph correspond to stable docking configurations, which
+are of interest in determining how the molecule interacts with the protein.
+
+The first step is to import the Strawberry Fields GBS module and external dependencies:
+"""
+from strawberryfields import gbs
+import numpy as np
+import networkx as nx
+import plotly
+
+##############################################################################
+# The adjacency matrix of the TACE-AS graph can be loaded from the data module and the
+# graph can be visualized using the :mod:`~gbs.plot` module:
+
+TA = gbs.data.TaceAs()
+A = TA.adj
+TA_graph = nx.Graph(A)
+graph = gbs.plot.plot_graph(TA_graph)
+plotly.offline.plot(graph, filename="TACE-AS.html")
+
+##############################################################################
+# .. raw:: html
+#     :file: ../../examples_gbs/TACE-AS.html
+
+##############################################################################
+# Can you spot any cliques in the graph? It's not so easy using only your eyes! The TACE-AS graph
+# is sufficiently small that all cliques can be found by performing an exhaustive search over
+# all subgraphs. For example, below we highlight a small *maximal* clique, i.e., a clique
+# not contained inside another clique:
+
+maximal_clique = [4, 11, 12, 18]
+clique_0 = gbs.plot.plot_graph(TA_graph, maximal_clique)
+plotly.offline.plot(clique_0, filename="maximal_clique.html")
+
+##############################################################################
+# .. raw:: html
+#     :file: ../../examples_gbs/maximal_clique.html
+
+##############################################################################
+# We'll now use the :mod:`~gbs.clique` module to find larger cliques in the graph. We can make
+# use of the pre-generated samples to post-select outputs with any specific number of clicks. For
+# this tutorial, we'll look at samples with eight clicks, of which there are a total of 1,984:
+
+TA_subgraphs = gbs.sample.to_subgraphs(TA_graph, TA)
+counts = TA.counts()
+samples = [TA_subgraphs[i] for i in range(len(TA)) if counts[i] == 8]
+print(len(samples))
+
+##############################################################################
+# GBS produces samples that correspond to subgraphs of high density. For fun, let's confirm this
+# by comparing the average subgraph density in the GBS samples to uniformly generated samples:
+
+GBS_dens = []
+u_dens = []
+
+for s in samples:
+    uniform = list(np.random.choice(24, 8))  # generates uniform sample
+    GBS_dens.append(nx.density(TA_graph.subgraph(s)))
+    u_dens.append(nx.density(TA_graph.subgraph(uniform)))
+
+print('GBS mean density = {:.4f}'.format(np.mean(GBS_dens)))
+print('Uniform mean density = {:.4f}'.format(np.mean(u_dens)))
+
+##############################################################################

```

Comment:
Strangely I got a density of 0.9405 (on multiple runs) for GBS this time 🤔 Not sure why it's different.

Ok I guess I'm satisfied anyway since the uniform density seems to hover around 0.58.

Reply:
Update: The difference is because we need to move the `postselect` function above the call to `to_subgraphs`